com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -531,14 +531,30 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager != null && NetworkManager.IsListening && NetworkManager.IsServer == false && IsSpawned &&
+            // If no NetworkManager is assigned, then just exit early
+            if (!NetworkManager)
+            {
+                return;
+            }
+
+            if (NetworkManager.IsListening && !NetworkManager.IsServer && IsSpawned &&
                 (IsSceneObject == null || (IsSceneObject.Value != true)))
             {
-                throw new NotServerException($"Destroy a spawned {nameof(NetworkObject)} on a non-host client is not valid. Call {nameof(Destroy)} or {nameof(Despawn)} on the server/host instead.");
```

Comment:
could this just simply be replaced with an error log without changing things around it too much?
if we're only trying to error-and-continue instead of exception-throw to keep the shutdown sequence running, let's "just" do that and nothing else as the minimal fix?